src/perl/Permabit/VolumeGroup.pm

```diff
@@ -457,13 +488,18 @@ sub getFreeBytes {
 sub getLVMConfig {
   my ($self, $config) = assertMinMaxArgs([""], 1, 2, @_);
   my $storageDevice = $self->{storageDevice};
-  my $filter = "devices {scan_lvs=1 use_devicesfile=0}";
-  if ($storageDevice->isa('Permabit::BlockDevice::TestDevice')) {
-    my $underlyingDevice = $storageDevice->getStorageDevice();
-    my $underlyingStorage = $underlyingDevice->getDevicePath();
-    $filter .= " devices {filter=r|$underlyingStorage|}";
+  my $filter = "devices/scan_lvs=1";
+  if ($config eq "") {
+    $config = $filter;
+  } else {
+    $config = $config . " $filter";
+  }
+  $config = "--config '" . $config . "'";
+  my $devices = $self->getLVMDevices();
+  if ($devices eq "") {
+    return $config;
   }
```

Comment:
Changing the input variable ($config) seems like really bad form. (I don't recall if it can have side-effects in perl.) Please use a local variable to construct the string.

The string construction itself is sort of awkward, too. What about, say, constructing a list of config arguments and using join(' ', @args)? (Extra spaces in the test command don't seem like that big a deal, either.)

Reply:
I've reworked the functions so that we have a new function called getLVMOptions which returns the combination of getLVMDevices and getLVMConfig. I've cleaned up getLVMConfig to make it much simpler. There is change for extra spaces now.